libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -94,21 +104,27 @@ export const RecommendationPanelContext = (props: RecommendationPanelContextProp
 
   const onOperationClick = useCallback(
     (id: string, apiId?: string) => {
-      const operation = (allOperations ?? []).find((o: DiscoveryOperation<DiscoveryResultTypes>) => {
-        return apiId ? o.id === id && o.properties?.api?.id === apiId : o.id === id;
+      const searchOperation = SearchService().getOperationById?.bind(SearchService());
+
+      const searchResultPromise = searchOperation
+        ? searchOperation(id)
+        : Promise.resolve((allOperations ?? []).find((o) => (apiId ? o.id === id && o.properties?.api?.id === apiId : o.id === id)));
```

Comment:
I think doing this as a `.find` over the `allOperations` array first, then if the result is null doing the `searchOperation` call (if available) would make more sense if we're going to support a more hybrid approach to prefetch/realtime searching, but until we want to get into all that I think this works better

Reply:
I don't really believe we should depend on the data for the codepath here. What if the data really is empty (no match)? or what if the operations were still loading? 

 I believe the better choice here is the explicit code choice: did the host implement that method or not.

Reply:
Talked offline, the possible future hybrid approach would need to be settled a bit first before we can choose what to do here. For now given the fork, this is the best approach.